lib/FusionInventory/Agent/Task/Inventory/Generic/Rudder.pm

```diff
@@ -42,19 +44,65 @@ sub doInventory {
     # Get agent capabilities
     my @agentCapabilities = _listAgentCapabilities();
 
+    my $customProperties = _getCustomProperties(logger => $logger);
+
     my $rudder = {
         HOSTNAME => $hostname,
         UUID => $Uuid,
         AGENT => \@agents,
         SERVER_ROLES => { SERVER_ROLE => \@serverRoles },
         AGENT_CAPABILITIES => { AGENT_CAPABILITY => \@agentCapabilities },
+        CUSTOM_PROPERTIES => $customProperties,
     };
 
     $inventory->addEntry(
         section => 'RUDDER', entry => $rudder
     );
 }
 
+sub _getCustomProperties {
+    my (%params) = @_;
+    my $logger   = $params{logger};
+
+    my $custom_properties_dir = ($OSNAME eq 'MSWin32') ? 'C:\Program Files\Rudder\local\hooks.d' : '/var/rudder/local/hooks.d';
+    my $custom_properties;
+    if (-d "$custom_properties_dir") {
+        my @custom_propertes_list = ();
+        opendir(DIR, $custom_properties_dir); # or die $!;
+        # List each file in the custom_properties directory, each files being a script
+        my @ordered_script_list = sort readdir(DIR);
+        while (my $file = shift @ordered_script_list) {
+            my $script_file = $custom_properties_dir . "/" . $file;
+            if (-f $script_file) {
+                next if ($file =~ m/^\./);
+                # Ignore non-executable file, or folders
+                next unless -x $script_file;
+                my $properties = qx($script_file);
+                my $exit_code = $? >> 8;
+                if ($exit_code > 0) {
+                    $logger->error("Script $script_file failed to run properly, with exit code $exit_code");
```

Comment:
By convention, don't assume $logger is defined while it was passed in %params. So add a `if $logger` at the end of line before `;`

Reply:
ha, this is something I was not aware of